CRM/Contact/BAO/Relationship.php

```diff
@@ -1736,6 +1736,10 @@ public static function relatedMemberships($contactId, &$params, $ids, $action =
               //contact before creating new membership record.
               CRM_Member_BAO_Membership::deleteRelatedMemberships($membershipId, $relatedContactId);
             }
+            //skip status calculation for pay later memberships.
+            if (!empty($membershipValues['is_pay_later']) && !empty($membershipValues['status_id'])) {
```

Comment:
I feel like it's the pendingness not the pay-later-ness that means we should skip the calc? That might be more reliable to check?

Reply:
Can't find any problems in modifying the above check to pending. I've updated the PR to look for the same now. Cheers 👍